src/utils/web.vala

```diff
@@ -76,7 +79,11 @@ namespace ProtonPlus.Utils {
                     bytes_downloaded += output_stream.write (chunk.get_data ());
 
                     if (progress_callback != null) {
-                        var progress = (int) (((double) bytes_downloaded / download_size) * 100);
+                        // Use static "50% Progress" when download size is unknown.
+                        // TODO: Enhance the progress callback so it can display
+                        // "1.13 MiB" (human-formatting "bytes_downloaded" to KiB
+                        // and MiB) if no filesize available? To show actual progress.
+                        var progress = download_size <= 0 ? 50 : (int) (((double) bytes_downloaded / download_size) * 100);
```

Comment:
I`d prefer only showing a spinner if the download size is not set..

Reply:
I just changed this because with the current math, it was showing download percentages such as "124823821385%". So I just forced it to 50% whenever size is unknown. And made a note that it could be enhanced to show "data downloaded so far" (instead of a percentage) when size is unknown.

I have not touched anything related to showing a spinner or not. If you want to change that (and to remove the "TODO" idea), I would suggest merging and then doing changing the spinner implementation. For example you might change this line to this:

```vala
var progress = download_size <= 0 ? -1 : (int) (((double) bytes_downloaded / download_size) * 100);
```

And then the progress callback can interpret `-1` as "no freaking idea!". :)

Btw: We DO NOT KNOW the download size AT ALL for the SteamTinkerLaunch downloads, because GitHub generates those live and streams the data as it's being zipped by the server in realtime. Size is not known. Which is why I had to fix this to at least show "50%" rather than "1381381283123%". :)

Reply:
PS: The spinner while downloading makes sense even when the download size is unknown. The "50% progress" workaround was just a temporary workaround, hence the "TODO" note. It's up to you to decide what to do there. For example, the progress callback could take 2 parameters (progress and is_percentage) which tells it to treat the "progress" value as either 0-100% progress or as a byte value.

Then, if progress is a byte value ("for unknown download sizes"), convert it to human-readable (the most basic would be to always convert to MiB with a static facor, and round it to 2 decimals), and display the value (instead of a percentage), such as "1.32 MiB", "1.65 MiB", etc...

Reply:
I just spotted that filesystem utils has a `covert_bytes_to_string()` function if you like the idea of converting the download bytes to a human readable progress bar.

Reply:
How would you display the progress if we used the way you mentioned?
Something like this 1.56/2.85
Or maybe something else?

Reply:
Sorry, GitHub again didn't send me any emails for this discussion. I'll answer that one in the thread instead though. :)